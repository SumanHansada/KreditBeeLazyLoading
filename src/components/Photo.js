import React, { useState, useEffect } from 'react';
import LazyLoad from 'react-lazy-load';
import { Card } from 'react-bootstrap';

function Photo(props) {
  const { albumId } = props;
  const [isLoading, setIsLoading] = useState(true);
  const [photos, setPhotos] = useState([]);

  useEffect(() => {
    fetch('http://jsonplaceholder.typicode.com/photos?albumId=' + albumId, {
      method: 'GET',
    })
      .then((res) => res.json())
      .then((response) => {
        setPhotos(response);
        setIsLoading(false);
      })
      .catch((error) => console.log(error));
  }, [albumId]);

  return (
    <div className="horizontal-scroll-div">
      {isLoading && <p>Loading Photos</p>}
      {photos.length > 0 &&
        photos.map((photo, index) => (
          <LazyLoad
            key={index}
            offsetHorizontal={50}
            throttle={500}
            debounce={true}
          >
            <Card className="card-container">
              <Card.Body>
                <LazyLoad offsetHorizontal={50} throttle={500} debounce={true}>
                  <Card.Img
                    alt={photo.title}
                    variant="top"
                    src={photo.thumbnailUrl}
                  />
                </LazyLoad>
                <Card.Text className="wrap-text">
                  {photo.title}
                  <br />
                  id: {photo.id}
                </Card.Text>
              </Card.Body>
            </Card>
          </LazyLoad>
        ))}
    </div>
  );
}

export default Photo;
