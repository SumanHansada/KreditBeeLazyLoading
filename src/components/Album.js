import React, { useEffect, useState } from "react";
import { Card } from "react-bootstrap";
import Photo from "./Photo";
import LazyLoad from "react-lazy-load";

function Album() {
  const [isLoading, setIsLoading] = useState(false);
  const [albums, setAlbums] = useState([]);

  useEffect(() => {
    setIsLoading(true);
    fetch("http://jsonplaceholder.typicode.com/albums", {
      method: "GET",
    })
      .then((res) => res.json())
      .then((response) => {
        setAlbums(response);
        setIsLoading(false);
      })
      .catch((error) => console.log(error));
  }, []);

  return (
    <div>
      {isLoading && <p>Loading Albums</p>}
      {albums.length > 0 &&
        albums.map((album, index) => (
          <LazyLoad
            key={index}
            offsetVertical={300}
            throttle={500}
            debounce={true}
          >
            <Card>
              <Card.Header>
                <Card.Title>{album.title}</Card.Title>
                <Card.Text>
                  id: {album.id}, userid: {album.userId}
                </Card.Text>
              </Card.Header>
              <LazyLoad throttle={500} debounce={true} offsetHorizontal={50}>
                <Photo albumId={album.id}></Photo>
              </LazyLoad>
            </Card>
          </LazyLoad>
        ))}
    </div>
  );
}

export default Album;
