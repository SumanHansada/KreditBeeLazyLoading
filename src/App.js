import "bootstrap/dist/css/bootstrap.min.css";
import "./App.css";
import Album from "./components/Album";

function App() {
  return (
    <div className="App">
      <Album />
    </div>
  );
}

export default App;
