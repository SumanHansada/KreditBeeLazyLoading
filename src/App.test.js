import toJson from 'enzyme-to-json';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import App from './App';
import Album from './components/Album';
import Photo from './components/Photo';
import { act } from 'react-dom/test-utils';
import React from 'react';

Enzyme.configure({ adapter: new Adapter() });

it('renders App component', () => {
  const AppComponent = shallow(<App />);
  expect(toJson(AppComponent)).toMatchSnapshot();
});

it('renders Photo component', () => {
  const PhotoComponent = shallow(<Photo />);
  expect(toJson(PhotoComponent)).toMatchSnapshot();
});

describe('Album', () => {
  it('should render', () => {
    const AlbumComponent = shallow(<Album />);
    expect(toJson(AlbumComponent)).toMatchSnapshot();
  });
  it('should fetch data after rendering', () => {
    let AlbumComponent;
    const albums = [];
    React.useState = jest.fn().mockReturnValue([
      albums,
      [
        {
          userId: 1,
          id: 1,
          title: 'quidem molestiae enim',
        },
        {
          userId: 1,
          id: 2,
          title: 'sunt qui excepturi placeat culpa',
        },
      ],
    ]);
    act(() => {
      AlbumComponent = mount(<Album />);
    });
    expect(toJson(AlbumComponent)).toMatchSnapshot();
  });
});
