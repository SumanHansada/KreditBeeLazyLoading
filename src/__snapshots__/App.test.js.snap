// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album should fetch data after rendering 1`] = `
<Album>
  <div>
    <p>
      Loading Albums
    </p>
  </div>
</Album>
`;

exports[`Album should render 1`] = `
<div>
  <p>
    Loading Albums
  </p>
</div>
`;

exports[`renders App component 1`] = `
<div
  className="App"
>
  <Album />
</div>
`;

exports[`renders Photo component 1`] = `
<div
  className="horizontal-scroll-div"
>
  <p>
    Loading Photos
  </p>
</div>
`;
